gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ui/SnappyRegionStatsCollectorFunction.java

```diff
@@ -145,6 +145,10 @@ private SnappyRegionStats collectDataFromBean(LocalRegion lr, RegionMXBean bean)
   private long getEntryOverhead(RegionEntry entry,
       GemFireXDInstrumentation sizer) {
     long entryOverhead = sizer.sizeof(entry);
+    Object key = entry.getRawKey();
+    if (key != null) {
+      entryOverhead += sizer.sizeof(key);
+    }
```

Comment:
This is going to be very inefficient with reflection walk-through for every key. In fact the whole call of getEntryOverhead() from PartitionedRegion should not be per entry. Instead what the loop does for replicated region is better option. Get the entry overhead once, then in the iterator call calcMemsize for key if null. So the calling point from PartitionRegion case should be made similar to ReplicatedRegion to fix this.

Reply:
@sumwale I don't think it walks through every entry. It just calls this method once per BucketRegion.

Reply:
@rishitesh Ok, but then how will it give correct result? Some keys may be null (non-overflowed ones) and some not. Its not a problem just for generated key case from what I can see from code.

Reply:
Replicate one does the correct thing. Loops through and adds calcMemsize() for non-null keys. Similar should be done for Partitioned case.

Reply:
@sumwale Ok. Will change it similar to replicated regions

Reply:
@sumwale Ok, probably this will need another stats in bucket region inline with inMemorySize. This stat will keep aggregate in memory key sizes of the region. We can add/remove to this stat during snapshot. That would be a more comprehensive solution.